### DISCUSSION TITLE
Proposal: new keyword uniqueValues

### DISCUSSION THREAD

**Discussion by SorinGFS** [json-schema-org/community#426] [Specification]
There is a situation where array of objects may conflict if some items has properties with identical values. E.g.,

```json
[
    {"name": "John"},
    {"name": "John"}
]
```

This could be invalidated by a keyword like `uniqueValues` which should be an array containing the property names which should be checked to contain unique values.

**Comment by gregsdennis** [comment]
I have this defined as a vocabulary already: https://docs.json-everything.net/schema/vocabs/uniquekeys/

**Reply by SorinGFS** [reply]
Your keyword has a specific use case. What I mean is different, let me be more specific. Let's consider the following schema portion:

```json
   "type": "array",
   "items": {
        "type": "object",
        "properties": {
            "email": { "type": "string" },
            "foo": { "type": "string" }
        },
        "uniqueValues": [ "email" ],
        "required": [ "email" ]
    }
```

The above schema would validate:

```json
[
    { "email": "a@a.com", "foo": "bar" },
    { "email": "b@a.com", "foo": "bar" }
]
```

...and would invalidate:

```json
[
    { "email": "a@a.com", "foo": "bar" },
    { "email": "a@a.com", "foo": "baz" }
]
```

**Reply by gregsdennis** [reply]
That's exactly what my vocab does.

**Reply by SorinGFS** [reply]
Indeed. I was initially confused by that kind of unintuitive referring to the key by its `json-pointer`.

**Reply by gregsdennis** [reply]
Duplicate of https://github.com/json-schema-org/json-schema-vocabularies/issues/22
